x/wasm/internal/keeper/keeper.go

```diff
@@ -566,6 +566,10 @@ func (k Keeper) dispatchMessages(ctx sdk.Context, contractAddr sdk.AccAddress, m
 
 func gasForContract(ctx sdk.Context) uint64 {
 	meter := ctx.GasMeter()
+	// avoid integer overflow
+	if meter.IsOutOfGas() {
```

Comment:
A better solution would be to use `meter.GasConsumedToLimit()` which is the proper API to avoid underflow: https://github.com/cosmos/cosmos-sdk/blob/v0.39.1/store/types/gas.go#L62-L67

However, looking at the code for the only two implementations of GasMeter, this will never fail. We cannot trust that I guess, but using `GasConsumedToLimit` uses an API that double-ensures we never return a value over `Limit`)